interface/java_binding/src/main/cpp/libktx-jni.cpp

```diff
@@ -97,12 +97,14 @@ void copy_ktx_astc_params(JNIEnv *env, jobject params, ktxAstcParams &out)
     out.normalMap = env->GetBooleanField(params, normalMap);
     out.perceptual = env->GetBooleanField(params, perceptual);
 
-    env->GetByteArrayRegion(
-        static_cast<jbyteArray>(env->GetObjectField(params, inputSwizzle)),
-        0,
-        4,
-        reinterpret_cast<jbyte*>(&out.inputSwizzle)
-    );
+    jobject inputSwizzleObject = env->GetObjectField(params, inputSwizzle);
+    jcharArray inputSwizzleArray = static_cast<jcharArray>(inputSwizzleObject);
+    jchar *inputSwizzleValues = env->GetCharArrayElements( inputSwizzleArray, NULL);
+    for (int i=0; i<4; i++)
+    {
+      out.inputSwizzle[i] = static_cast<char>(inputSwizzleValues[i]);
```

Comment:
A static_cast will convert a 16-bit jchar to an 8-bit char? Seriously!

Reply:
At this point, this is basically the same as a `out = (char)in` C-Style cast. 

One could probably throw in some Unicode character into the swizzle array, where the _lower-order_ bits correspond to that of the `r` character. 

And we could think about certain forms of error checks here. For example, the KTX tools already do check the swizzle chars against the regex `"^[rgba01]{4}$"`, and we could insert similar checks at different places. This could happen in the JNI layer or (far more easily) in the Java layer. But error handling is one of the broader topics to be sorted out...

Reply:
Yes we need to resolve how to do error handling first. That should tell us whether to do it in JNI or Java. I'll mark this as resolved since it doesn't directly affect this PR.